pkg/beacon/analyzer.go

```diff
@@ -273,15 +274,14 @@ func (a *analyzer) hostIcertQuery(icert bool, src string, dst string) updateInfo
 
 		newFlag := false
 
-		type hostRes struct {
-			IP string `bson:"ip"`
-		}
+		var resList []interface{}
 
-		var resList []hostRes
+		hostSelector := src.BSONKey()
+		hostSelector["dat.icdst"] = dst.BSONKey()
 
-		_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.Structure.HostTable).Find(bson.M{"ip": src, "dat.icdst": dst}).All(&resList)
+		_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.Structure.HostTable).Find(hostSelector).All(&resList)
 
-		if !(len(resList) > 0) {
```

Comment:
Good call on changing this one...the previous way of stating "less than or equal to zero" was...strange...

Reply:
There were other places in the code I noticed this hadn't yet been changed. We should make an effort to switch them all eventually.